src/main/kotlin/org/rust/cargo/runconfig/CargoRunStateBase.kt

```diff
@@ -26,13 +26,15 @@ abstract class CargoRunStateBase(
     val runConfiguration: CargoCommandConfiguration,
     val config: CargoCommandConfiguration.CleanConfiguration.Ok
 ) : CommandLineState(environment) {
-    private val toolchain: RustToolchain = config.toolchain
+    val toolchain: RustToolchain = config.toolchain
     val commandLine: CargoCommandLine = config.cmd
     val cargoProject: CargoProject? = CargoCommandConfiguration.findCargoProject(
         environment.project,
         commandLine.additionalArguments,
         commandLine.workingDirectory
     )
+    private val workingDirectory = cargoProject?.workingDirectory
+    val computeSysroot: () -> String? = { workingDirectory?.let { toolchain.getSysroot(it) } }
```

Comment:
Why did you convert `computeSysroot` into property?
Let's revert this change and use function reference syntax instead